src/Base.php

```diff
@@ -377,11 +377,12 @@ protected function getOutPatternObj($grad, $objref)
      */
     public function getOutGradientShaders($pon)
     {
+        $this->pon = (int) $pon;
```

Comment:
The "pon" is intentionally not updated if the document is PDF/A or there are no gradients.
Do you have a test or a specific case that explains why this has to be changed?

Reply:
I created script like this:
```
$pdf = new Tcpdf();
$pdf->page->add();
$pdf->setFont('pdfacourier'); // method adds new font to stack / I can pull request it, but it is not finished yet
echo $pdf->getOutPDFString();
```

In output printed the PON number is duplicated for objects after graph.

In Output.php:
```
        $out .= $this->graph->getOutGradientShaders($this->pon); // pon is correct when goes into method
        $this->pon = $this->graph->getObjectNumber(); // pon is set to value lower than it was before, which causes a problem with PON duplication
```

So if you say it should not be updated, therefore I can suggest two other solutions different than this one.
1. Do not reassign the pon from method $this->graph->getObjectNumber() in the same case what the pon is not updated in ```getOutGradientShaders```
2. Set a higher number max(oldPon, getObjectNumber), therefore it will always get higher numer.